From anton@ozlabs.org Fri Dec 16 00:54:56 2005
Received: (at submit) by bugs.debian.org; 16 Dec 2005 08:54:57 +0000
Return-path: <anton@ozlabs.org>
Received: from ozlabs.org ([203.10.76.45])
	by spohr.debian.org with esmtp (Exim 4.50)
	id 1EnBMe-0003w7-M2
	for submit@bugs.debian.org; Fri, 16 Dec 2005 00:54:56 -0800
Received: by ozlabs.org (Postfix, from userid 1010)
	id AEC0E688AF; Fri, 16 Dec 2005 19:54:55 +1100 (EST)
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
From: Anton Blanchard <anton@samba.org>
To: Debian Bug Tracking System <submit@bugs.debian.org>
Subject: libc6: FTBFS [powerpc] weak alias issues
X-Mailer: reportbug 3.17
Date: Fri, 16 Dec 2005 19:43:41 +1100
Message-Id: <20051216085455.AEC0E688AF@ozlabs.org>
Delivered-To: submit@bugs.debian.org
X-Spam-Checker-Version: SpamAssassin 2.60-bugs.debian.org_2005_01_02 
	(1.212-2003-09-23-exp) on spohr.debian.org
X-Spam-Level: 
X-Spam-Status: No, hits=-8.0 required=4.0 tests=BAYES_00,HAS_PACKAGE 
	autolearn=no version=2.60-bugs.debian.org_2005_01_02

Package: libc6
Version: 2.3.5-8
Severity: serious
Tags: patch
Justification: no longer builds from source


I was unable to build the latest version of glibc on an up to date
powerpc unstable system. The following patch fixes it for me. The fixes
come straight out of mainline glibc.

--- sysdeps/powerpc/powerpc32/fpu/s_lround.S~	2004-06-30 17:29:12.000000000 -0500
+++ sysdeps/powerpc/powerpc32/fpu/s_lround.S	2005-11-18 16:48:11.000000000 -0600
@@ -88,7 +79,6 @@
 	b	.L9
 	END (__lround)
 
-strong_alias (__lround, __lround)
 weak_alias (__lround, lround)
 
 strong_alias (__lround, __lroundf)
--- sysdeps/unix/sysv/linux/powerpc/powerpc32/socket.S~	2005-12-16 01:04:21.000000000 -0600
+++ sysdeps/unix/sysv/linux/powerpc/powerpc32/socket.S	2005-12-16 01:06:26.000000000 -0600
@@ -42,7 +42,11 @@
 #define stackblock 20
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 	.text
@@ -116,4 +120,6 @@
 
 PSEUDO_END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
--- sysdeps/unix/sysv/linux/powerpc/powerpc64/socket.S~	2005-12-16 01:08:45.000000000 -0600
+++ sysdeps/unix/sysv/linux/powerpc/powerpc64/socket.S	2005-12-16 01:09:32.000000000 -0600
@@ -41,12 +41,12 @@
 
 #define stackblock 80 /* offset to socket parm area.  */
 
-#ifndef socket
-/* If this is just socket.S leave it alone! */
-#else
 #ifndef __socket
-#define __socket P(__,socket)
-#endif
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 	.text
@@ -120,4 +120,6 @@
 	cfi_endproc
 PSEUDO_END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif


-- System Information:
Debian Release: testing/unstable
  APT prefers unstable
  APT policy: (500, 'unstable')
Architecture: powerpc (ppc)
Shell:  /bin/sh linked to /bin/bash
Kernel: Linux 2.6.15-rc2-g5d240918
Locale: LANG=en_AU, LC_CTYPE=en_AU (charmap=ISO-8859-1)

-- no debconf information



